docs/src/main/asciidoc/deploying-to-google-cloud.adoc

```diff
@@ -0,0 +1,226 @@
+
+=== Building the application via gcloud
+
+You can choose to let `gcloud` build your application for you, this is the simplest way to deploy to App Engine.
+
+Just launch `gcloud app deploy` on the root of your project, it will upload all your project files (the list can be reduced via the  `.gcloudignore` file),
+package your JAR via Maven (or Gradle) and launch it on App Engine.
+
+When done, the output will display the URL of your application (target url), you can use it with curl or directly open it on your browser using `gcloud app browse`.
+
+=== Building the application via the Maven Google Cloud plugin
+
+You can also let Maven control the deployment of your application using the App Engine Maven plugin.
+
+First, add the plugin to your `pom.xml`:
+
+[source, xml]
+----
+<plugin>
+    <groupId>com.google.cloud.tools</groupId>
+    <artifactId>appengine-maven-plugin</artifactId>
+    <version>2.4.0</version>
+    <configuration>
+        <projectId>GCLOUD_CONFIG</projectId> <!-- <1> -->
+        <version>gettingstarted</version>
+        <artifact>${project.build.directory}/getting-started-${project.version}-runner.jar</artifact> <!-- <2> -->
+    </configuration>
+</plugin>
+----
+
+1. Use the default `gcloud` configuration
+2. Override the default JAR name to the one generated by the Quarkus Maven plugin
+
+Then you would be able to use Maven to build and deploy your application to App Engine via `mvn clean package appengine:deploy`.
+
+When it’s done, the output will display the URL of your application (target URL), you can use it with curl or directly open it on your browser using `gcloud app browse`.
+
+== Deploying to Google App Engine Flexible Custom Runtimes
+
+Before all, make sure to have an App Engine environment initialized for your Google Cloud project, if not, initialize one via `gcloud app create --project=[YOUR_PROJECT_ID]`.
+
+App Engine Flexible Custom Runtimes uses a Docker image to run your application.
+
+First, create an `app.yaml` file at the root of your project with the following content:
+
+[source, yaml]
+----
+runtime: custom
+env: flex
+----
+
+App Engine Flexible Custom Runtimes deploys your application as a Docker container, you can choose to deploy one of the Dockerfile provided inside your application.
+
+Both JVM and native executable versions will work.
+
+To deploy a JVM application:
+
+- Copy the JVM Dockerfile to the root directory of your project: `cp src/main/docker/Dockerfile.jvm Dockerfile`.
+- Build your application using `mvn clean package`.
+
+To deploy a native application:
+
+- Copy the native Dockerfile to the root directory of your project: `cp src/main/docker/Dockerfile.native Dockerfile`.
+- Build your application as a native executable using `mvn clean package -Pnative`.
+
+Finally, launch `gcloud app deploy` on the root of your project, it will upload all your project files (the list can be reduced via the  `.gcloudignore` file),
```

Comment:
```suggestion
Finally, launch `gcloud app deploy` in the root of your project, it will upload all your project files (the list can be reduced via the  `.gcloudignore` file),
```